File: euler_044.py
from time import time
from math import sqrt


def is_integer(f):
    return f-int(f) == 0.0

def is_pentagonal(i):
    return is_integer((1 + sqrt(1 + 24*i))/6)

start = time()

pentagonals = [n for n in range(1,10000000) if is_pentagonal(n)]

difference = set()

for i in range(len(pentagonals)):
    for j in range(i, len(pentagonals)):
        if is_pentagonal(pentagonals[j]-pentagonals[i]):
            difference.add((pentagonals[i], pentagonals[j]))

sums = []

for d in difference:
    if is_pentagonal(d[0] + d[1]):
        sums.append(d)

print("Pair: " + str(sums) + ", D: " + str(sums[0][1]-sums[0][0]))

stop = time()
print("Time: " + str(stop-start))

File: euler_044.js
const start = Date.now();

let D = 0;

// The quest to find any pentagon differences
let pentagonals = [1];
let n = 2;
let p = 5;

while (p < 10**7) {
    pentagonals.push(p);
    p += (3 * n + 1);
    n++;
}

let diffPairs = [];

for (let i = 0; i < pentagonals.length; i++) {
    for (let j = i; j < pentagonals.length; j++) {
        let diff = pentagonals[j] - pentagonals[i];
        if (pentagonals.indexOf(diff) > -1) {
            diffPairs.push([pentagonals[j],pentagonals[i]]);
        }
    }    
}

let sumAndDiffPairs = [];

for (const pair of diffPairs) {
    let sum = pair[0] + pair[1];
        if (pentagonals.indexOf(sum) > -1) {
            sumAndDiffPairs.push(pair);
        }
}

console.log(sumAndDiffPairs[0][0] - sumAndDiffPairs[0][1]);


const end = Date.now();

console.log("Execute time: ", end - start, " The value of D is: ", D);